hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/address_16c/TokenTypeTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.suites.contract.precompile.address_16c;
+
+import static com.hedera.services.bdd.junit.TestTags.SMART_CONTRACT;
+import static com.hedera.services.bdd.spec.HapiSpec.hapiTest;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecContract.VARIANT_16C;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecTokenKey.FEE_SCHEDULE_KEY;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecTokenKey.FREEZE_KEY;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecTokenKey.KYC_KEY;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecTokenKey.PAUSE_KEY;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecTokenKey.SUPPLY_KEY;
+import static com.hedera.services.bdd.spec.dsl.entities.SpecTokenKey.WIPE_KEY;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.overriding;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.HapiTestLifecycle;
+import com.hedera.services.bdd.spec.dsl.annotations.Contract;
+import com.hedera.services.bdd.spec.dsl.annotations.FungibleToken;
+import com.hedera.services.bdd.spec.dsl.entities.SpecContract;
+import com.hedera.services.bdd.spec.dsl.entities.SpecFungibleToken;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Tag;
+
+@Tag(SMART_CONTRACT)
+@DisplayName("tokenType")
+@HapiTestLifecycle
+public class TokenTypeTest {
```

Comment:
So here I would like to suggest:  You're setting the new standard here for how this is versioning of an _overload_ is to be done.  

So I think you should take the original `TokenAndTypeCheckSuite.java` and move it to a `address_16c` package, and move its corresponding `TokenAndTypeCheck.sol` contract to a `contracts_167` resource directory.  And I guess that means splitting up the original suite and contract so that `getTokenType` which is now varianted (is that a word? well, it is now) is separate from `isToken` which is not.  

And I know this is a extra work in this PR and future override-method PRs but if it isn't done then IMO things will just get very confusing later on as more override-methods happen.  

(But as usual it is your call!)

Reply:
Good suggestion.

Reply:
(I meant of course _override_ not overload, sorry!)